src/backend/commands/matview.c

```diff
@@ -4256,3 +4267,45 @@ isIvmColumn(const char *s)
 {
 	return (strncmp(s, "__ivm_", 6) == 0);
 }
+
+/*
+ * get_securityQuals
+ *
+ * Get row security policy on a relation.
+ * This is used by IVM for copying RLS from base table to enr.
+ */
+static List *
+get_securityQuals(Oid relId,Query *query)
+{
+	ParseState *pstate;
+	Relation rel;
+	RangeTblEntry *rte;
+	List *securityQuals;
+	List *withCheckOptions;
+	bool  hasRowSecurity;
+	bool  hasSubLinks;
+
+	securityQuals = NIL;
+	pstate = make_parsestate(NULL);
+
+	rel = table_open(relId, NoLock);
+	rte = addRangeTableEntryForRelation(pstate, rel, AccessShareLock, NULL, false, false);
+
+	get_row_security_policies(query, rte, 1,
+							  &securityQuals, &withCheckOptions,
+							  &hasRowSecurity, &hasSubLinks);
+	rte->securityQuals = list_concat(securityQuals, rte->securityQuals);
```

Comment:
Why is this list_concat necessary? It seems that this RTE is not used at all, although I might be missing something, though.

Reply:
> > @thoshiai Could you please resolve the conflicts?
> 
> I resolved conflict

Thanks!

Reply:
>Why is this list_concat necessary? It seems that this RTE is not used at all, although I might be missing something, though.

After some investigation, I think this list_concat is unnecessary.  Maybe you borrowed this code from fireRIRrules() which needs to rewrite RTE, but get_securityQuals() just needs to get the security guals of a base table for ENR ,and RTE of a base table is not necessary.  Actually, the regression test works well even if we comments this list_concat out.

Reply:
OK, I think so, too.

Reply:
Sorry , I missed it.